UIKitTemplate/UIKitTemplate/Controllers/FeedViewController.swift

```diff
@@ -0,0 +1,124 @@
+// FeedViewController.swift
+// Copyright © RoadMap. All rights reserved.
+
+import UIKit
+
+/// Лента постов
+final class FeedViewController: UIViewController {
+    // MARK: - Constants
+
+    private enum FeedSectionType {
+        case post(Bool)
+        case stories
+        case recommendation
+    }
+
+    // MARK: - Visual Components
+
+    private lazy var feedTableView: UITableView = {
+        let table = UITableView()
+        table.delegate = self
+        table.dataSource = self
+        table.register(StoriesCell.self, forCellReuseIdentifier: StoriesCell.reuseID)
+        table.register(PostCell.self, forCellReuseIdentifier: PostCell.reuseID)
+        table.register(RecommendationCell.self, forCellReuseIdentifier: RecommendationCell.reuseID)
+        table.separatorStyle = .none
+        table.rowHeight = UITableView.automaticDimension
+        return table
+    }()
+
+    private lazy var refreshControl: UIRefreshControl = {
+        let refreshControl = UIRefreshControl()
+        refreshControl.addTarget(self, action: #selector(handleRefresh(_:)), for: .valueChanged)
+        return refreshControl
+    }()
+
+    // MARK: - Private Properties
+
+    private let sections = [FeedSectionType.stories, .post(true), .recommendation, .post(false)]
+
+    // MARK: - Life Cycle
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        setupNavigationItem()
+        setupView()
+    }
+
+    // MARK: - Private Methods
+
+    private func setupView() {
+        feedTableView.addSubview(refreshControl)
+        view.addSubview(feedTableView)
+
+        setupContraints()
+    }
+
+    private func setupContraints() {
+        feedTableView.translatesAutoresizingMaskIntoConstraints = false
+        [
+            feedTableView.topAnchor.constraint(equalTo: view.safeAreaLayoutGuide.topAnchor),
+            feedTableView.leadingAnchor.constraint(equalTo: view.leadingAnchor),
+            feedTableView.trailingAnchor.constraint(equalTo: view.trailingAnchor),
+            feedTableView.bottomAnchor.constraint(equalTo: view.safeAreaLayoutGuide.bottomAnchor)
+        ].activate()
+    }
+
+    private func setupNavigationItem() {
+        let messagesBarItem = UIBarButtonItem(image: .messagesBarIcon.withRenderingMode(.alwaysOriginal))
+        let logoBarItem = UIBarButtonItem(image: .logo.withRenderingMode(.alwaysOriginal))
+        navigationItem.setRightBarButton(messagesBarItem, animated: false)
+        navigationItem.setLeftBarButton(logoBarItem, animated: false)
+    }
+
+    @objc private func handleRefresh(_ refreshControl: UIRefreshControl) {
+        refreshControl.endRefreshing()
+    }
+}
+
+// MARK: - UITableViewDelegate
+
+extension FeedViewController: UITableViewDelegate {}
+
+// MARK: - UITableViewDataSource
```

Comment:
```suggestion
// MARK: - FeedViewController + UITableViewDataSource
```
 делегаты и датасорсы так оформляем